modules/charon-core/src/main/java/org/wso2/charon3/core/utils/AttributeUtil.java

```diff
@@ -170,22 +170,37 @@ public static Boolean parseBoolean(Object booleanValue) throws BadRequestExcepti
      * Will iterate through <code>{@code SCIMAttributeSchema}</code> objects
      *
      * @param attributeName
+     * @param schema
      * @return
      */
     public static String getAttributeURI(String attributeName, SCIMResourceTypeSchema schema) throws
             BadRequestException {
 
+        return getAttributeURI(attributeName, schema, false);
+    }
+
+    /*
+     * Will iterate through <code>{@code SCIMAttributeSchema}</code> objects
+     *
+     * @param attributeName
+     * @param schema
+     * @param isFilter
+     * @return
+     */
+    public static String getAttributeURI(String attributeName, SCIMResourceTypeSchema schema, boolean isFilter) throws
+            BadRequestException {
+
         Iterator<AttributeSchema> attributeSchemas = schema.getAttributesList().iterator();
         while (attributeSchemas.hasNext()) {
             AttributeSchema attributeSchema = attributeSchemas.next();
 
-            if (attributeSchema.getName().equals(attributeName) || attributeSchema.getURI().equals(attributeName)) {
+            if ((isFilter && attributeSchema.getName().equalsIgnoreCase(attributeName)) ||
+                    attributeSchema.getName().equals(attributeName) || attributeSchema.getURI().equals(attributeName)) {
```

Comment:
attributeSchema.getName().equals(attributeName) would be redundant with attributeSchema.getName().equalsIgnoreCase(attributeName))